shared/common-adapters/input.native.js

```diff
@@ -262,6 +262,10 @@ class Input extends Component<Props, State> {
       ...(this.props.maxLength ? {maxlength: this.props.maxLength} : null),
     }
 
+    if (this.props.uncontrolled) {
+      delete commonProps['value']
```

Comment:
Sorry, I think this is partially my fault. My refactoring PR added `state.value = props.value || ''` in the constructor, which I think makes it always controlled. A more robust fix might be to revert that change.

Reply:
But yeah maybe we want a more explicit split between uncontrolled and controlled text input.